unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/RPCSocialApiBridge.cs

```diff
@@ -0,0 +1,238 @@
+            //
+            // await foreach (var item in stream)
+            // {
+            //     // this.friends.Add("asd", new User() { Address = "asd" });
+            //
+            //     // this.OnAddNewFriend?.Invoke(item);
+            // }
+        }
+
+        public async UniTask RejectFriendship(string friendRequestId, CancellationToken cancellationToken = default)
+        {
+            string userId = GetUserIdFromFriendRequestId(friendRequestId);
+
+            var updateFriendshipPayload = new UpdateFriendshipPayload() { Event = EventType.Reject, User = new User() { Address = userId } };
+
+            await this.UpdateFriendship(updateFriendshipPayload, cancellationToken);
+
+            OnFriendRequestRemoved?.Invoke(userId);
+        }
+
+        public async UniTask<FriendRequest> RequestFriendship(string friendUserId, string messageBody, CancellationToken cancellationToken = default)
+        {
+            long createdAt = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
+
+            var updateFriendshipPayload = new UpdateFriendshipPayload()
+            {
+                Event = EventType.Request, Request = new Decentraland.Social.Friendships.Request()
+                {
+                    Message = messageBody, User = new User() { Address = friendUserId }, CreatedAt = createdAt
+                }
+            };
+
+            await UpdateFriendship(updateFriendshipPayload, cancellationToken);
+
+            return new FriendRequest(GetFriendRequestId(friendUserId, createdAt), createdAt, "GET OWN ADDRESS", friendUserId, messageBody);
+        }
+
+        private string GetUserIdFromFriendRequestId(string friendRequestId) =>
+            friendRequestId.Split("-")[0];
+
+        private string GetFriendRequestId(string userId, long createdAt) =>
+            $"{userId}-{createdAt}";
+
+        private async UniTask UpdateFriendship(UpdateFriendshipPayload updateFriendshipPayload, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                // TODO: pass cancellation token to rpc client when is supported
+                var response = await rpc.Social()
+                                        .UpdateFriendshipEvent(updateFriendshipPayload)
+                                        .Timeout(TimeSpan.FromSeconds(REQUEST_TIMEOUT));
+
+                switch (response.ReplyCase)
+                {
+                    case UpdateFriendshipResponse.ReplyOneofCase.Event:
+                        break;
+                    case UpdateFriendshipResponse.ReplyOneofCase.Error:
+                        var error = response.Error;
+                        throw new FriendshipException(ToErrorCode(error));
+                    case UpdateFriendshipResponse.ReplyOneofCase.None:
+                    default:
+                        throw new InvalidCastException();
```

Comment:
instead could be `NotSupportedException` passing `response.ReplyCase` in the message